packages/beacon-node/src/sync/range/chain.ts

```diff
@@ -123,9 +144,16 @@ export class SyncChain {
     this.processChainSegment = fns.processChainSegment;
     this.downloadBeaconBlocksByRange = fns.downloadBeaconBlocksByRange;
     this.reportPeer = fns.reportPeer;
+    this.getConnectedPeerSyncMeta = fns.getConnectedPeerSyncMeta;
     this.config = modules.config;
+    this.custodyConfig = modules.custodyConfig;
     this.logger = modules.logger;
-    this.logId = `${syncType}`;
+    this.metrics = modules.metrics;
+    this.logId = `${syncType}-${nextChainId++}`;
+
+    if (this.metrics != null) {
+      this.metrics.syncRange.headSyncPeers.addCollect(() => this.scrapeMetrics(this.metrics as Metrics));
```

Comment:
why is the type cast needed?

```suggestion
      this.metrics.syncRange.headSyncPeers.addCollect(() => this.scrapeMetrics(this.metrics));
```

Reply:
updated in #8164